gemrb/core/Scriptable/Actor.cpp

```diff
@@ -8569,22 +8568,20 @@ void Actor::SetSoundFolder(const ieVariable& soundset) const
 		return;
 	}
 
-	char filepath[_MAX_PATH];
-
 	PCStats->SoundFolder = soundset;
-	PathJoin(filepath, core->config.GamePath, "sounds", PCStats->SoundFolder.c_str(), nullptr);
+	path_t filepath = PathJoin(core->config.GamePath, "sounds", PCStats->SoundFolder);
 
 	DirectoryIterator dirIt(filepath);
 	dirIt.SetFilterPredicate(new EndsWithFilter("01"));
 	dirIt.SetFlags(DirectoryIterator::Directories);
 	if (dirIt) {
 		do {
-			const char* name = dirIt.GetName();
-			const char* end = strchr(name, '.');
-			if (end != nullptr) {
+			const path_t& name = dirIt.GetName();
+			size_t end = FindFirstOf(name, ".");
+			if (end != path_t::npos) {
 				// need to truncate the "01" from the name, eg. HaFT_01.wav -> HaFT
 				// but also 2df_007.wav -> 2df_0, meaning the data is less diverse than it may seem
-				PCStats->SoundSet.Format("{:.{}}", name, int(end - 2 - name));
```

Comment:
BTW, I don't understand this. To me, it looks like we are casting a pointer to an int. that cant possibly be right can it?

Reply:
oh, I guess this controls the field width?

Reply:
Yep. We could use erase instead, now that it's a string.